integration_tests/web/test_admin_conversations_restrictAccess.py

```diff
@@ -0,0 +1,84 @@
+import asyncio
+import logging
+import os
+import unittest
+
+from integration_tests.env_variable_names import SLACK_SDK_TEST_GRID_WORKSPACE_ADMIN_USER_TOKEN, \
+    SLACK_SDK_TEST_GRID_ORG_ADMIN_USER_TOKEN, \
+    SLACK_SDK_TEST_GRID_IDP_USERGROUP_ID, SLACK_SDK_TEST_GRID_TEAM_ID
+from integration_tests.helpers import async_test
+from slack import WebClient
+
+
+class TestWebClient(unittest.TestCase):
+    """Runs integration tests with real Slack API"""
+
+    def setUp(self):
+        self.logger = logging.getLogger(__name__)
+        self.org_admin_token = os.environ[SLACK_SDK_TEST_GRID_ORG_ADMIN_USER_TOKEN]
+        self.sync_client: WebClient = WebClient(
+            token=self.org_admin_token,
+            run_async=False,
+            loop=asyncio.new_event_loop()
+        )
+        self.async_client: WebClient = WebClient(token=self.org_admin_token, run_async=True)
+
+        self.team_id = os.environ[SLACK_SDK_TEST_GRID_TEAM_ID]
+        self.idp_group_id = os.environ[SLACK_SDK_TEST_GRID_IDP_USERGROUP_ID]
+
+        if not hasattr(self, "channel_id"):
+            team_admin_token = os.environ[SLACK_SDK_TEST_GRID_WORKSPACE_ADMIN_USER_TOKEN]
+            client = WebClient(token=team_admin_token, run_async=False, loop=asyncio.new_event_loop())
+            # Only fetching private channels since admin.conversations.restrictAccess methods do not work for public channels
+            convs = client.conversations_list(exclude_archived=True, limit=100, types="private_channel")
+            self.channel_id = next((c["id"] for c in convs["channels"] if c["name"] == "general"), None)
+
+    def tearDown(self):
+        pass
+
+    def test_sync(self):
+        client = self.sync_client
+
+        add_group = client.admin_conversations_restrictAccess_addGroup(
+            channel_id=self.channel_id,
+            group_id=self.idp_group_id,
+            team_id=self.team_id
+        )
+        self.assertIsNotNone(add_group)
+
```

Comment:
As they're in Tier 1 and 2, can we have some `time.sleep()` to avoid rate-limited errors? Checking the Java SDK's test code would be helpful. https://github.com/slackapi/java-slack-sdk/pull/507/files#diff-d5e0cb7cfc4938a8a74df14dbc3577fa